tools/makerootfs.sh

```diff
@@ -56,6 +56,37 @@ do_imagefromtar() {
   cat "${tarfile}" | docker run -i --rm -v /dev:/dev --privileged -v "$IMAGE:/rootfs.img" "${MKROOTFS_TAG}"
 }
 
+abspath() {
+  local target
+  local dir
+  local filename
+  local absolute_dir
+
+  target="$1"
+  if [ -z "$target" ]; then
+    echo "Error: Unable to find file '$target'" >&2
+    return 1
+  fi
+  # we have two problems here:
+  # First, we want to realpath a file that may not exist yet.
+  # on some OSes, it is fine; on others, it returns an error.
+  # a prerequisite is that the directory exists, so we can realpath that,
+  # and just append the filename after.
+  # Second, we want to use realpath, but it is not available on all OSes.
+  dir=$(dirname "$target")
+  filename=$(basename "$target")
+  if [ ! -d "$dir" ]; then
+    echo "Error: Unable to find directory '$dir'" >&2
+    return 1
+  fi
+  if command -v realpath >/dev/null 2>&1; then
+    absolute_dir=$(realpath "$dir")
+  else
+    absolute_dir=$(cd "$(dirname "$target")"; pwd -P)
```

Comment:
You can use $dir here instead of redoing dirname.

Reply:
Fixed